src/Misc/externals.sh

```diff
@@ -221,10 +221,20 @@ if [[ "$PACKAGERUNTIME" == "linux-arm64" ]]; then
 fi
 
 if [[ "$PACKAGERUNTIME" != "win-x64" && "$PACKAGERUNTIME" != "win-x86" ]]; then
-    rm -rf "$LAYOUT_DIR/externals/node/lib/node_modules/npm"
+    # remove `npm`, `npx`, `corepack`, and related `node_modules` from the `externals/node*` agent directory
+    # they are installed along with node, but agent does not use them
+
+    rm -rf "$LAYOUT_DIR/externals/node/lib"
     rm "$LAYOUT_DIR/externals/node/bin/npm"
-    rm -rf "$LAYOUT_DIR/externals/node10/lib/node_modules/npm"
+
+    rm -rf "$LAYOUT_DIR/externals/node10/lib"
     rm "$LAYOUT_DIR/externals/node10/bin/npm"
+    rm "$LAYOUT_DIR/externals/node10/bin/npx"
+
+    rm -rf "$LAYOUT_DIR/externals/node16/lib"
+    rm "$LAYOUT_DIR/externals/node16/bin/npm"
+    rm "$LAYOUT_DIR/externals/node16/bin/npx"
+    rm "$LAYOUT_DIR/externals/node16/bin/corepack"
```

Comment:
It would be nice if we did a pre-check of the existence of the files that will be deleted. They aren't always present.
![image](https://user-images.githubusercontent.com/93121155/178158824-50a5c9ab-2746-4128-8bba-7d174e0aebed.png)


Like this
```
if [ -f "$LAYOUT_DIR/externals/node16/bin/corepack" ]; then
    rm "$LAYOUT_DIR/externals/node16/bin/corepack"
fi
```

Reply:
[These messages](https://dev.azure.com/mseng/AzureDevOps/_build/results?buildId=17659150&view=logs&j=f4485888-349a-5114-5431-db0c31b0a7f9&t=644b6e19-cacf-50c0-b439-cf1d5eb56ef7&l=109) are displayed when removing symlinks, but they are misleading.
In fact, all the symlinks are successfully located and removed.
Tried using `unlink` instead of `rm`, but got the same result.

Reply:
I checked and discussed it with Denis. These messages do not affect the agent, and my suggestions do not apply here.